### PR TITLE
Modified all the hardcoded GCS URIs to dl.k8s.io

### LINKED ISSUE #1551: Use dl.k8s.io instead of hardcoded GCS URIs
https://github.com/kubernetes-sigs/cluster-api/pull/4958

/help

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1551 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by sayantani11** [comment]
@CecileRobertMichon I had to clone the project again to use the ``` make generate ``` command, from cloning to everything I did in WSL, and it worked fine.
Thanks to @marosset he told me to this, and hence I had to create another branch, and hence another PR.
Sorry for all the mix up, it's my first contribution so sorry :')

**Comment by sayantani11** [comment]
I just hope this works

File: templates/test/ci/cluster-template-prow-ci-version.yaml [hunk 1]
```diff
@@ -137,21 +130,20 @@ spec:
               DEBIAN_FRONTEND=noninteractive apt-get install -y $$CI_PACKAGE=$$PACKAGE_VERSION
             done
           else
-            CI_URL="gs://k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
-            if ! $${GSUTIL} ls "$${CI_URL}" > /dev/null; then
-              CI_URL="gs://k8s-release-dev/ci/$$CI_VERSION/bin/linux/amd64"
+            CI_URL="dl.k8s.io/k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/cluster-template-prow-ci-version.yaml:133]:
```suggestion
```

Comment (CecileRobertMichon) [reply templates/test/ci/cluster-template-prow-ci-version.yaml:133]: we only want the line below

File: templates/test/ci/prow-machine-pool-ci-version/patches/machine-pool-ci-version.yaml [hunk 2]
```diff
@@ -54,21 +47,20 @@ spec:
               DEBIAN_FRONTEND=noninteractive apt-get install -y $$CI_PACKAGE=$$PACKAGE_VERSION
             done
           else
-            CI_URL="gs://k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
-            if ! $${GSUTIL} ls "$${CI_URL}" > /dev/null; then
-              CI_URL="gs://k8s-release-dev/ci/$$CI_VERSION/bin/linux/amd64"
+            CI_URL="dl.k8s.io/k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/prow-machine-pool-ci-version/patches/machine-pool-ci-version.yaml:50]:
```suggestion
```

Comment (CecileRobertMichon) [reply templates/test/ci/prow-machine-pool-ci-version/patches/machine-pool-ci-version.yaml:50]: we only want the line below

File: templates/test/ci/prow-ci-version/patches/machine-deployment-ci-version.yaml [hunk 3]
```diff
@@ -56,21 +49,20 @@ spec:
                   DEBIAN_FRONTEND=noninteractive apt-get install -y $$CI_PACKAGE=$$PACKAGE_VERSION
                 done
               else
-                CI_URL="gs://k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
-                if ! $${GSUTIL} ls "$${CI_URL}" > /dev/null; then
-                  CI_URL="gs://k8s-release-dev/ci/$$CI_VERSION/bin/linux/amd64"
+                CI_URL="dl.k8s.io/k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/prow-ci-version/patches/machine-deployment-ci-version.yaml:52]:
```suggestion
```

Comment (CecileRobertMichon) [reply templates/test/ci/prow-ci-version/patches/machine-deployment-ci-version.yaml:52]: same here

File: templates/test/ci/patches/control-plane-ci-version.yaml [hunk 4]
```diff
@@ -58,21 +51,20 @@ spec:
               DEBIAN_FRONTEND=noninteractive apt-get install -y $$CI_PACKAGE=$$PACKAGE_VERSION
             done
           else
-            CI_URL="gs://k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
-            if ! $${GSUTIL} ls "$${CI_URL}" > /dev/null; then
-              CI_URL="gs://k8s-release-dev/ci/$$CI_VERSION/bin/linux/amd64"
+            CI_URL="dl.k8s.io/k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/patches/control-plane-ci-version.yaml:54]:
```suggestion
```

Comment (CecileRobertMichon) [reply templates/test/ci/patches/control-plane-ci-version.yaml:54]: same here

File: templates/test/ci/patches/control-plane-ci-version.yaml [hunk 5]
```diff
@@ -20,14 +20,7 @@ spec:
         set -o pipefail
         set -o errexit
         [[ $(id -u) != 0 ]] && SUDO="sudo" || SUDO=""
-        GSUTIL=gsutil
```
Comment (CecileRobertMichon) [inline templates/test/ci/patches/control-plane-ci-version.yaml:23]: please also change `CI_URL="gs://kubernetes-release/release/$$CI_VERSION/bin/linux/amd64"` on line 47 to `CI_URL="https://storage.googleapis.com/kubernetes-release/release/$${KUBERNETES_VERSION}/bin/linux/amd64"`

Comment (sayantani11) [reply templates/test/ci/patches/control-plane-ci-version.yaml:23]: Is it for every file?

Comment (CecileRobertMichon) [reply templates/test/ci/patches/control-plane-ci-version.yaml:23]: yes

**Review by CecileRobertMichon — COMMENTED** [review]
/ok-to-test

**Comment by sayantani11** [comment]
@CecileRobertMichon can you tell me what's the error?

**Comment by CecileRobertMichon** [comment]
unfortunately, it failed to fetch logs so I can't see the error, going to rerun

/retest

**Comment by sayantani11** [comment]
@CecileRobertMichon Okay

**Comment by CecileRobertMichon** [comment]
just found there's a bug in our e2e log collection that was introduced yesterday, fixing that right now so we can get logs again

**Comment by CecileRobertMichon** [comment]
fix merged

/retest

File: templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml [hunk 6]
```diff
@@ -123,7 +116,7 @@ spec:
           echo "* testing CI version $$CI_VERSION"
           # Check for semver
           if [[ "$${CI_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-            CI_URL="gs://kubernetes-release/release/$$CI_VERSION/bin/linux/amd64"
+            CI_URL="https://storage.googleapis.com/kubernetes-release/release/$${KUBERNETES_VERSION}/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml:119]:
```suggestion
            CI_URL="https://storage.googleapis.com/kubernetes-release/release/$${CI_VERSION}/bin/linux/amd64"
```

We're not defining `KUBERNETES_VERSION` anywhere, our variable is called `CI_VERSION`, this is what is causing the test to fail with `/tmp/kubeadm-bootstrap.sh: line 38: KUBERNETES_VERSION: unbound variable`

same thing applies to the other files

File: templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml [hunk 7]
```diff
@@ -137,21 +130,19 @@ spec:
               DEBIAN_FRONTEND=noninteractive apt-get install -y $$CI_PACKAGE=$$PACKAGE_VERSION
             done
           else
-            CI_URL="gs://k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
-            if ! $${GSUTIL} ls "$${CI_URL}" > /dev/null; then
-              CI_URL="gs://k8s-release-dev/ci/$$CI_VERSION/bin/linux/amd64"
+              CI_URL="https://storage.googleapis.com/k8s-release-dev/ci/$${KUBERNETES_VERSION}/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml:133]:
```suggestion
              CI_URL="https://storage.googleapis.com/k8s-release-dev/ci/$${CI_VERSION}/bin/linux/amd64"
```

File: templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml [hunk 8]
```diff
@@ -313,7 +297,7 @@ spec:
         echo "* testing CI version $$CI_VERSION"
         # Check for semver
         if [[ "$${CI_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-          CI_URL="gs://kubernetes-release/release/$$CI_VERSION/bin/linux/amd64"
+          CI_URL="https://storage.googleapis.com/kubernetes-release/release/$${KUBERNETES_VERSION}/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml:300]:
```suggestion
          CI_URL="https://storage.googleapis.com/kubernetes-release/release/$${CI_VERSION}/bin/linux/amd64"
```

File: templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml [hunk 9]
```diff
@@ -327,21 +311,19 @@ spec:
             DEBIAN_FRONTEND=noninteractive apt-get install -y $$CI_PACKAGE=$$PACKAGE_VERSION
           done
         else
-          CI_URL="gs://k8s-release-dev/ci-periodic/$$CI_VERSION/bin/linux/amd64"
-          if ! $${GSUTIL} ls "$${CI_URL}" > /dev/null; then
-            CI_URL="gs://k8s-release-dev/ci/$$CI_VERSION/bin/linux/amd64"
+          CI_URL="https://storage.googleapis.com/k8s-release-dev/ci/$${KUBERNETES_VERSION}/bin/linux/amd64"
```
Comment (CecileRobertMichon) [inline templates/test/ci/cluster-template-prow-machine-pool-ci-version.yaml:314]:
```suggestion
          CI_URL="https://storage.googleapis.com/k8s-release-dev/ci/$${CI_VERSION}/bin/linux/amd64"
```

**Comment by CecileRobertMichon** [comment]
ci-version test is passing now

/retest
e2e test flake

/cc @jsturtevant

**Pull request merged**